src/token/TokenRejectTransaction.js

```diff
@@ -0,0 +1,270 @@
+ */
+
+/**
+ * Reject a new Hedera™ crypto-currency token.
+ */
+export default class TokenRejectTransaction extends Transaction {
+    /**
+     *
+     * @param {object} [props]
+     * @param {AccountId | string | null} [props.owner]
+     * @param {HashgraphProto.proto.TokenReference[]} [props.rejections]
+     */
+    constructor(props = {}) {
+        super();
+
+        /**
+         * @private
+         * @type {?AccountId}
+         */
+        this._owner = null;
+
+        if (props.owner != null) {
+            this.setOwnerId(props.owner);
+        }
+
+        /**
+         * @private
+         * @type {TokenId[]}
+         */
+        this._tokenIds = [];
+
+        /**
+         * @private
+         * @type {NftId[]}
+         */
+        this._nftIds = [];
+
+        if (props.rejections != null) {
+            for (let rejection of props.rejections) {
+                if (rejection instanceof TokenId) {
+                    this._tokenIds.push(rejection);
+                } else if (rejection instanceof NftId) {
+                    this._nftIds.push(rejection);
+                }
+            }
+        }
+    }
+
+    /**
+     * @internal
+     * @param {HashgraphProto.proto.ITransaction[]} transactions
+     * @param {HashgraphProto.proto.ISignedTransaction[]} signedTransactions
+     * @param {TransactionId[]} transactionIds
+     * @param {AccountId[]} nodeIds
+     * @param {HashgraphProto.proto.ITransactionBody[]} bodies
+     * @returns {TokenRejectTransaction}
+     */
+    static _fromProtobuf(
+        transactions,
+        signedTransactions,
+        transactionIds,
+        nodeIds,
+        bodies,
+    ) {
+        const body = bodies[0];
+        const rejectToken =
+            /** @type {HashgraphProto.proto.ITokenRejectTransactionBody} */ (
+                body.tokenReject
+            );
+
+        return Transaction._fromProtobufTransactions(
+            new TokenRejectTransaction({
+                owner:
+                    rejectToken.owner != null
+                        ? AccountId._fromProtobuf(rejectToken.owner)
+                        : undefined,
+                rejections: rejectToken.rejections
+                    ? rejectToken.rejections
+                    : [],
+            }),
+            transactions,
+            signedTransactions,
+            transactionIds,
+            nodeIds,
+            bodies,
+        );
+    }
+
+    /**
+     * @returns {TokenId[] | string[]}
+     * @readonly
+     */
+    get tokenIds() {
+        return this._tokenIds;
+    }
+
+    /**
+     * @param {TokenId[] | string[]} tokenIds
+     * @returns {this}
+     */
+    setTokenIds(tokenIds) {
+        this._requireNotFrozen();
+        this._tokenIds = tokenIds.map((tokenId) =>
+            tokenId instanceof TokenId ? tokenId : TokenId.fromString(tokenId),
+        );
+        return this;
+    }
+
+    /**
+     * @param {TokenId | string} tokenId
```

Comment:
Again why the string? This is valid for other places as well.

Reply:
Well if it's some common practice I guess you can do that but I don't see a reason for you having it either way. If the user has a `string[]` he could just use `fromString` to build his tokenIds. I don't think other SDKs allow this type of behavior.